tests/unit/request/node.ts

```diff
@@ -1,23 +1,193 @@
+	expectedData?: any;
+	expectedToError?: boolean;
+	followRedirects?: boolean;
+	callback?: (_: any) => void;
+	keepOriginalMethod?: boolean;
+}
+
+const responseData: { [url: string]: DummyResponse } = {
+	'foo.json': {
+		body: JSON.stringify({ foo: 'bar' })
+	},
+	invalidJson: {
+		body: '<not>JSON</not>'
+	},
+	'redirect-success': {
+		body: JSON.stringify({ success: true })
+	},
+	'300-redirect': {
+		statusCode: 300,
+		body: JSON.stringify('beginning to redirect'),
+		headers: {
+			'Location': getRequestUrl('redirect-success')
+		}
+	},
+	'301-redirect': {
+		statusCode: 301,
+		body: JSON.stringify('beginning to redirect'),
+		headers: {
+			'Location': getRequestUrl('redirect-success')
+		}
+	},
+	'302-redirect': {
+		statusCode: 302,
+		body: JSON.stringify('beginning to redirect'),
+		headers: {
+			'Location': getRequestUrl('redirect-success')
+		}
+	},
+	'303-redirect': {
+		statusCode: 303,
+		body: JSON.stringify('beginning to redirect'),
+		headers: {
+			'Location': getRequestUrl('redirect-success')
+		}
+	},
+	'304-redirect': {
+		statusCode: 304,
+		body: JSON.stringify('beginning to redirect'),
+		headers: {
+			'Location': getRequestUrl('redirect-success')
+		}
+	},
+	'305-redirect': {
+		statusCode: 305,
+		body: JSON.stringify('beginning to redirect'),
+		headers: {
+			'Location': 'http://localhost:1337'
+		}
+	},
+	'305-redirect-broken': {
+		statusCode: 305,
+		body: JSON.stringify('beginning to redirect')
+	},
+	'306-redirect': {
+		statusCode: 306,
+		body: JSON.stringify('beginning to redirect'),
+		headers: {
+			'Location': getRequestUrl('redirect-success')
+		}
+	},
+	'307-redirect': {
+		statusCode: 307,
+		body: JSON.stringify('beginning to redirect'),
+		headers: {
+			'Location': getRequestUrl('redirect-success')
+		}
+	},
+	'infinite-redirect': {
+		statusCode: 301,
+		body: JSON.stringify('beginning to redirect'),
+		headers: {
+			'Location': getRequestUrl('infinite-redirect')
+		}
+	},
+	'broken-redirect': {
+		statusCode: 301,
+		body: JSON.stringify('beginning to redirect'),
+		headers: {}
+	}
+};
+
+function buildRedirectTests(methods: RedirectTestData[]) {
+	let tests: { [key: string]: () => void } = {};
+
+	methods.forEach((details) => {
+		const method = details.method;
+		const { keepOriginalMethod = false } = details;
+		const url = getRequestUrl(details.url);
+		const expectedMethod = details.expectedMethod || method;
+		const expectedPage = details.expectedPage ? getRequestUrl(details.expectedPage) : url;
+		const followRedirects = (details.followRedirects === undefined) ? true : details.followRedirects;
+
+		let title = details.title || (method + ' ' + (keepOriginalMethod ? 'w/' : 'w/o') + ' keepOriginalMethod');
+
+		tests[ title ] = () => {
+
+			let error: any = null;
+
+			return nodeRequest(getRequestUrl(details.url), {
+				responseType: 'json',
+				method: method,
+				followRedirects: followRedirects,
+				redirectOptions: {
+					keepOriginalMethod
+				}
+			}).then((response) => {
```

Comment:
We need to be more explicit about this in the style guide, but chains like this should linebreak before the next method... So:

```
})
.then((response) => {
```